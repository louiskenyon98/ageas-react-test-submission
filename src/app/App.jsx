import React from 'react';
import Routes from './routes';
import ModalContainer from './containers/common/Modal';

class App extends React.Component {

    render() {
        //Check error prop and only render content if there are no api errors.
        //Using low-level interface "<Router>" to sync custom history.
        return (
            <React.Fragment>
                <Routes/>
                <ModalContainer/>
            </React.Fragment>
        )
    }
}

export default App;
