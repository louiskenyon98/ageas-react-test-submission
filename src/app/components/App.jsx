import React from 'react';
import {Router, Route} from 'react-router-dom';
import {connect} from 'react-redux';

import Navigation from './Navigation';
import history from '../history';

import {getNotes} from '../actions';
import AllNotes from './AllNotes';
import NoteCreate from './NoteCreate';
import NoteEdit from './NoteEdit';

class App extends React.Component {
    componentDidMount() {
        //Calls getNotes function so as to perform error checking functionality.
        this.props.getNotes();
    }
    render() {
        //Check error prop and only render content if there are no api errors.
        //Using low-level interface "<Router>" to sync custom history.
        if(this.props.error === false) {
            return (
                <div>
                    <Router history={history}>
                        <div>
                            <Navigation/>
                            <Route path="/" exact component={AllNotes}/>
                            <Route path="/new" exact component={NoteCreate}/>
                            <Route path="/edit/:id" exact component={NoteEdit}/>
                        </div>
                    </Router>
                </div>
            )
        }
        return (
            <div>
                <Router history={history}>
                    <div>
                        <Navigation/>
                        <div className="ui icon negative message">
                            <i className="frown outline icon"/>
                            <div className="content">
                                <div className="header">
                                    <h3>Sorry, there has been an error contacting the API.</h3>
                                </div>
                            </div>
                        </div>
                    </div>
                </Router>
            </div>
        )
    }
}
const mapStateToProps = (state) => {
    return {
        error: state.error
    }
};
export default connect(mapStateToProps, {getNotes})(App);
