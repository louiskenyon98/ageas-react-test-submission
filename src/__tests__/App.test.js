import React from 'react';
import App from '../app/App';
import {shallow} from 'enzyme';

describe('App', () => {
    let props;
    let wrapper = () => shallow(<App {...props}/>);
    beforeEach(() => {
        props = {};
    });
    it('should render the Routes and ModalContainer components', () => {
        expect(wrapper()).toMatchSnapshot();
    })
});
