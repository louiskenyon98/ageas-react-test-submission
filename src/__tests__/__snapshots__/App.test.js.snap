// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the Routes and ModalContainer components 1`] = `
<Fragment>
  <Routes />
  <Connect(ModalContainer) />
</Fragment>
`;
